pkg/manager/config/proxy.go

```diff
@@ -24,46 +24,154 @@ import (
 	"go.uber.org/zap"
 )
 
+type CfgType int
+
+const (
+	CfgServer CfgType = iota
+	CfgLog
+)
+
+type OnlineCfgTypes interface {
+	config.ProxyServerOnline | config.LogOnline
+}
+
+type imeta interface {
+	getPrefix() string
+	unmarshal(bytes []byte) (any, error)
+	addToCh(any)
+	getInitial(cfg *config.Config) any
+}
+
+type meta[T OnlineCfgTypes] struct {
+	prefix   string
+	initFunc func(cfg *config.Config) T
+	ch       chan *T
+}
+
+func newMeta[T OnlineCfgTypes](prefix string, initFunc func(cfg *config.Config) T) *meta[T] {
+	return &meta[T]{
+		prefix:   prefix,
+		initFunc: initFunc,
+		ch:       make(chan *T, 1),
+	}
+}
+
+func (m meta[T]) unmarshal(bytes []byte) (any, error) {
+	var t T
+	err := json.Unmarshal(bytes, &t)
+	return &t, err
+}
+
+func (m meta[T]) getPrefix() string {
+	return m.prefix
+}
+
+func (m meta[T]) addToCh(obj any) {
+	m.ch <- obj.(*T)
+}
+
+func (m meta[T]) getInitial(cfg *config.Config) any {
+	return m.initFunc(cfg)
+}
+
+func (e *ConfigManager) initMetas() {
+	e.metas = map[CfgType]imeta{
+		CfgServer: newMeta(pathPrefixProxyServer, func(cfg *config.Config) config.ProxyServerOnline {
+			return cfg.Proxy.ProxyServerOnline
+		}),
+		CfgLog: newMeta(pathPrefixLog, func(cfg *config.Config) config.LogOnline {
+			return cfg.Log.LogOnline
+		}),
+	}
+}
+
 func (e *ConfigManager) watchCfgProxy(ctx context.Context, cfg *config.Config) error {
-	if _, err := e.GetProxyConfig(ctx); err != nil && errors.Is(err, ErrNoOrMultiResults) {
-		if err := e.SetProxyConfig(ctx, &config.ProxyServerOnline{
-			MaxConnections: cfg.Proxy.MaxConnections,
-			TCPKeepAlive:   cfg.Proxy.TCPKeepAlive,
-		}); err != nil {
+	for _, m := range e.metas {
+		if err := func(m imeta) error {
+			_, err := e.get(ctx, m.getPrefix(), "config")
+			if err != nil && errors.Is(err, ErrNoOrMultiResults) {
+				value, err := json.Marshal(m.getInitial(cfg))
+				if err != nil {
+					return err
+				}
+				if err = e.set(ctx, m.getPrefix(), "config", string(value)); err != nil {
```

Comment:
I want to change the prefix settings, it can be done in this PR, or another.

Previously, we do `/config/xxx/config`. I considered the suffix is for compatibility/reserving usages. But actually we could do `/config/v2/xxx` or `/v2/config/xxx` anyway. Let us just remove this useless suffix.